raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -99,7 +99,7 @@
         public final InetSocketAddress address;
 
         public InetAddressSpec(InetSocketAddress address) {
-            if (address.equals(NON_ROUTABLE_ADDRESS)) {
+            if (address != null && address.equals(NON_ROUTABLE_ADDRESS)) {
```

Comment:
We also don't want to accept null addresses, right? I was thinking we could do this:
```java
if (address == null || address.equals(NON_ROUTABLE_ADDRESS)) {
  throw new IllegalArgumentException("Invalid address " + address);
}
```

Reply:
Currently our equals method assumes non-null addresses. I can't think of a case where we would want it.

Reply:
Right. Makes sense. Fixed.